Feature/modeling and vis and tracking (#1)
snowplow/consent-accelerator: content/visualisation/visualisation_1.md

+++
title = "Streamlit"
post = ""
weight = 1
+++

> Streamlit uses Python to build shareable dashboards without the need for front-end development experience.

Reviewer: Page currently not displaying any content here on the Netlify site.

Reviewer: Jack has resolved this

```bash
pipenv install
pipenv shell
```

#### **Step 2:** Set-up Database Connection
Open `secrets.toml` and add your BigQuery account and database details.

{{% notice warning %}}
Ensure `secrets.toml` is in `.gitignore` to keep your information safe.
{{% /notice %}}

```toml
# .streamlit/secrets.toml

[gcp_service_account]
type = "xxx"
project_id = "xxx"
private_key_id = "xxx"
private_key = "xxx"
client_email = "xxx"
client_id = "xxx"
auth_uri = "xxx"
token_uri = "xxx"
auth_provider_x509_cert_url = "xxx"
client_x509_cert_url = "xxx"

```
#### **Step 3:** Specify schema
Fill out the schema variable within `Healthcheck_board.py` file. Make sure you specify your custom `derived` schema which will be the source schema for the dashboard.

#### **Step 4:** Run the Streamlit dashboard
Run the command below to run the streamlit locally

```bash
streamlit run Healthcheck_board.py
```

{{% notice tip %}}

> In case the dashboard does not load due to errors such as 'This session does not have a current database. Call 'USE DATABASE', or use a qualified name.' a possible workaround is to assign default ROLE to the Snowflake user that could handle this.'

Reviewer: The tip is referring to Snowflake, but we're in the Big Query tab here. Should this reference Big Query instead?

Author: Good spot! I will remove this block altogether as this error message is Snowflake specific.

> In case the dashboard does not load due to errors such as 'This session does not have a current database. Call 'USE DATABASE', or use a qualified name.' a possible workaround is to assign default ROLE to the Snowflake user that could handle this.'

Reviewer 2: ```suggestion
If the dashboard does not load due to errors such as 'This session does not have a current database. Call 'USE DATABASE', or use a qualified name.' a possible workaround is to assign default ROLE to the Snowflake user that could handle this.'
```

```bash
pipenv install
pipenv shell
```

#### **Step 2:** Set-up Database Connection
Open `secrets.toml` and add your Snowflake account and database details. Make sure you specify your custom `derived` schema which will be the source schema for the dashboard.

{{% notice warning %}}
Ensure `secrets.toml` is in `.gitignore` to keep your information safe.
{{% /notice %}}

```toml
# .streamlit/secrets.toml

[snowflake]
user = "xxx"
password = "xxx"
account = "xxx"
database = "xxx"
schema = "xxx"
warehouse = "xxx"
role = "xxx"

```
#### **Step 3:** Run the Streamlit dashboard
Run the command below to run the streamlit locally

```bash
streamlit run Healthcheck_board.py
```

{{% notice tip %}}
In case the dashboard does not load due to errors such as 'This session does not have a current database. Call 'USE DATABASE', or use a qualified name.' a possible workaround is to assign default ROLE to the Snowflake user that could handle this.'
{{% /notice %}}

{{% /tab %}}
{{< /tabs >}}

You will be able to see an interactive dashboard similar to this:

> 

Reviewer: Can we see if we can put these images in a figma design so that there is a web browser boarder around them?
